From 25290953c590da7c4c026b6f7ae22c14a7ffa04f Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 16 Nov 2019 05:06:39 +0200
Subject: [PATCH] [SW-6506] hl: remove unused symbols from binary

This will reduce binary size and hence shorten its copy time from flash
to sram by BTL

This is needed for warm reset of Goya where limit is 20mS and we are
just above it

Note:
Goya size reduced from 8312B to 4188B
Gaudi size reduced from 10760B to 6095B

Change-Id: I88475b23adc880de8eebc7e82c4c8c6dd269378e
---

diff --git a/Makefile b/Makefile
index 2259398..4212f3e 100644
--- a/Makefile
+++ b/Makefile
@@ -14,10 +14,12 @@
 	CFLAGS += -DTARGET_IS_TESTER
 endif
 
+CFLAGS += -Os -fdata-sections -ffunction-sections
 SFLAGS := $(CFLAGS)
 
 LD_SCRIPT := $(HL_DEV)/preboot.lds
 LDFLAGS := -g -T $(LD_SCRIPT) -nostdlib -static
+LDFLAGS += -Wl,--gc-sections
 LDLIBS := -lgcc
 
 include $(HL_DEV)/Makefile
